components/measure-upload/MeasureRepositoryUpload.tsx

```diff
@@ -0,0 +1,235 @@
+import { Button, Collapse, Grid, Group, Loader, Select, TextInput } from '@mantine/core';
+import React, { useState } from 'react';
+import { useEffect } from 'react';
+import { useRecoilState, useRecoilValue, useSetRecoilState } from 'recoil';
+import { measureBundleState } from '../../state/atoms/measureBundle';
+import { measurementPeriodState } from '../../state/atoms/measurementPeriod';
+import { identifyMissingValueSets, MeasureUploadProps, populateMeasurementPeriod } from '../../util/measureUploadUtils';
+import { showNotification } from '@mantine/notifications';
+import { v4 as uuidv4 } from 'uuid';
+import { IconAlertCircle } from '@tabler/icons';
+import { CircleCheck } from 'tabler-icons-react';
+import { displayMapToSelectDataState } from '../../state/selectors/displayMapToSelectData';
+
+enum PACKAGE_STATES {
+  LOADING,
+  SUCCESS,
+  FAIL,
+  NONE
+}
+
+const MEASURE_RETRIEVAL_BODY = {
+  resourceType: 'Parameters',
+  parameter: [
+    {
+      name: 'include-terminology',
+      valueBoolean: true
+    }
+  ]
+};
+
+const MEASURE_RETRIEVAL_OPTIONS = {
+  method: 'POST',
+  body: JSON.stringify(MEASURE_RETRIEVAL_BODY),
+  headers: { 'Content-Type': 'application/json+fhir' }
+};
+
+export default function MeasureRepositoryUpload({ logError }: MeasureUploadProps) {
+  const [isLoadingIds, setIsLoadingIds] = useState(false);
+  const [isLoadingPackage, setIsLoadingPackage] = useState<PACKAGE_STATES>(PACKAGE_STATES.NONE);
+  const [measureSelectIsOpen, setMeasureSelectIsOpen] = useState(false);
+
+  const [{ measureRepositoryUrl, isFile, selectedMeasureId, displayMap }, setMeasureBundle] =
+    useRecoilState(measureBundleState);
+  const setMeasurementPeriod = useSetRecoilState(measurementPeriodState);
+
+  const idSelectData = useRecoilValue(displayMapToSelectDataState);
+
+  useEffect(() => {
+    if (isFile) {
+      setMeasureSelectIsOpen(false);
+      setIsLoadingPackage(PACKAGE_STATES.NONE);
+    }
+  }, [isFile]);
+
+  const rejectUpload = (
+    message: string | string[],
+    attemptedBundleDisplay: string | null,
+    isValueSetMissingError = false
+  ) => {
+    logError({
+      id: uuidv4(),
+      message,
+      timestamp: new Date().toISOString(),
+      attemptedBundleDisplay,
+      isValueSetMissingError
+    });
+    showNotification({
+      icon: <IconAlertCircle />,
+      title: 'Bundle upload failed',
+      message: 'There was an issue uploading your bundle. Please correct the issues listed below.',
```

Comment:
The error message for measure _file_ upload is: "There was an issue with your file." Should we change the error message in `MeasureFileUpload.tsx` to match this one (since technically you are uploading a bundle there too...), or do you think we should keep the two messages separate to better differentiate what measure upload method caused the error? If you think we should keep two different messages, maybe we can add more context to this one and say something like "There was an issue uploading your bundle _from the server_"

Reply:
Hmm I'm leaning towards it being ok to have the same error message here, which is simplest when consolidating the two `rejectUpload/File` functions